components/float-button/style/index.tsx

```diff
@@ -0,0 +1,136 @@
+import type { CSSObject } from '@ant-design/cssinjs';
+import type { FullToken, GenerateStyle } from '../../theme';
+import { genComponentStyleHook, mergeToken } from '../../theme';
+import { resetComponent } from '../../style';
+
+/** Component only token. Which will handle additional calculation of alias token */
+export interface ComponentToken {
+  zIndexPopup: number;
+}
+
+type FloatButtonToken = FullToken<'FloatButton'> & {
+  floatButtonColor: string;
+  floatButtonBackgroundColor: string;
+  floatButtonHoverBackgroundColor: string;
+  floatButtonFontSize: number;
+  floatButtonSize: number;
+
+  // Position
+  floatButtonBlockEnd: number;
+  floatButtonInlineEnd: number;
+  floatButtonInlineEndMD: number;
+  floatButtonInlineEndXS: number;
+};
+
+// ============================== Shared ==============================
+const sharedFloatButtonStyle: GenerateStyle<FloatButtonToken, CSSObject> = (token): CSSObject => {
+  const { componentCls, floatButtonFontSize, floatButtonSize, zIndexPopup } = token;
+
+  return {
+    [componentCls]: {
+      ...resetComponent(token),
+      // position: 'fixed',
+      insetInlineEnd: token.floatButtonInlineEnd,
+      insetBlockEnd: token.floatButtonBlockEnd,
+      zIndex: zIndexPopup,
+      width: 40,
+      height: 40,
+      cursor: 'pointer',
+      overflow: 'hidden',
+      boxShadow: token.boxShadowSecondary,
+      '&:empty': {
+        display: 'none',
+      },
+      [`${componentCls}-circle`]: {
+        borderRadius: token.radiusBase * 20,
+      },
+      [`${componentCls}-square`]: {
+        height: 'auto',
+        minHeight: floatButtonSize,
+        borderRadius: token.radiusBase,
+      },
+      [`${componentCls}-default`]: {
+        backgroundColor: token.colorTextLightSolid,
+      },
+      [`${componentCls}-primary`]: {
+        backgroundColor: token.colorPrimaryHover,
```

Comment:
```suggestion
        backgroundColor: token.colorPrimary,
```
应该 hover 后才是 `colorPrimaryHover`